poolakey-rx/src/main/java/com/phelat/poolakey/rx/PaymentRx.kt

```diff
@@ -22,12 +22,16 @@ import io.reactivex.Single
  * @return Observable that you can subscribe to it and get notified about service connection changes.
  */
 fun Payment.connect(): Observable<Connection> {
+    var connection: Connection? = null
     return Observable.create<Connection> { emitter ->
         connect {
+            connection = this
```

Comment:
You can assign `connection` directly with the return value of `connect` function:
```kotlin
connection = connect {
    ...
}
```